src/br/com/caelum/cdc/servlet/UniqueAuthorNameValidator.java

```diff
@@ -0,0 +1,36 @@
+package br.com.caelum.cdc.servlet;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import br.com.caelum.cdc.dao.AuthorDao;
+
+public class UniqueAuthorNameValidator {
+
+	private AuthorDao authorDao;
+	private boolean isInvalid;
+	private static Map<String, String> error = new HashMap<>();
+	
+	public UniqueAuthorNameValidator() {
+		this.authorDao = new AuthorDao();
+	}
+
+	public void checkUniqueKey(String name) {
+		if (authorDao.findByName(name)) {
+			setInvalid(true);
```

Comment:
o método privado gera uma indireção aqui para uma lógica simples... podia mandar um this.isInvalid = true direto :)